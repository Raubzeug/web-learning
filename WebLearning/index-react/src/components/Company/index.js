import React from 'react';
import './company.less'

const Company = ({name, img}) => (
    <div className="company">
        <a className="company-logo" href="/#">
            <img src={img} alt={name}/>
        </a>
        <div className="company-name">
            {name}
        </div>
    </div>
)

export default Company