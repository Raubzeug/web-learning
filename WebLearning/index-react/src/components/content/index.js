import React from 'react';
import Partners from './Partners.js'
import Courses from './Courses.js';
import Reviews from './Reviews.js'


const ContentIndex = () => (
    <section className="content content__two-columns">
        <div className="content-left-wrapper">
            <div className="content__block">
                <div className="content__header">
                    Популярные курсы
                </div>
                <Courses />
            </div>
            <Reviews />
        </div>
        <Partners />

    </section>
)

export default ContentIndex