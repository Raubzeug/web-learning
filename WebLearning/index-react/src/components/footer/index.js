import React from 'react';

const Footer = () => (
    <div className="footer-wrapper">
        <div className="footer">
        ПОДВАЛ
        </div>
    </div>
)

export default Footer