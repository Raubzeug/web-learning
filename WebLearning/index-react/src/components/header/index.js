import React from 'react';
import Logo from '../Logo'
import Navigation from '../Navigation'
import Search from '../Search'
import UserControls from '../user-controls'
import './header.less'

const Header = () => (
    <nav className='header'>
        <Logo />
        <Navigation />
        <div className='header__right-wrapper'>
            <Search />
            <UserControls />
        </div>
    </nav>
)

export default Header;