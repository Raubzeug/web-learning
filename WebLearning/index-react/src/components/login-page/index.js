import React from 'react';
import { Redirect } from 'react-router'
import LoginForm from './LoginForm'
import {getCookie} from '../../js/getCookie'
import './login-content.less'


class LoginContent extends React.Component {
    constructor(props) {
        super(props)
        this.state = {success: '', error: '', errors: [], redirect: false}
        this.submitForm = this.submitForm.bind(this)
    }
    
    submitForm = (eventData) => {
        this.setState({success: '', error: '', errors: []})
        const headers = {
            'Content-Type': 'application/json',
            'Accept': 'application/json',
            'X-CSRFToken': getCookie('csrftoken'),
            }
        fetch('/api/auth/login/', {
            method: 'POST',
            headers: headers,
            credentials: 'include',
            body: JSON.stringify(eventData),
            })
            .then(response => {
                return response.json()})
                .then(data => {
                    if ('token' in data) {
                        this.setState({
                            success: 'You are sucessfully logged in!',
                            redirect: true
                        })
                    }
                    else {
                        for (var elem in data) {
                            this.setState({error: data[elem]})
                            this.setState(state => {
                                const errors = state.errors.concat(state.error)
                                return {
                                    errors,
                                    error: ''
                                }
                            })
                        }
                        }
                })
                .catch(err => console.error(err))        
    }

    render = () => {
        const redirect = this.state.redirect
        if (redirect) {
            return (
                <Redirect to={{
                    pathname: "/profile",
                    state: { username: this.state.success }
             }} />
            )
        }
        else {
            return (
                <section className="content">
                    <div className="content__header_gradient">
                            <span>Войти в личный кабинет</span>
                    </div>
                    <div className='centered-div-padding-10'>
                        Не зарегистрированы?<br/>
                        <a className='base-link' href='./registration'>Зарегистрироваться</a>
                    </div>
                    <LoginForm submit={this.submitForm} success={this.state.success}
                             errors={this.state.errors} />
                    <div className='centered-div-padding-10'>
                        Забыли пароль?<br/>
                        <a className='base-link' href='/'>Восстановление пароля</a>
                    </div>
                </section>
    )}}
}

export default LoginContent