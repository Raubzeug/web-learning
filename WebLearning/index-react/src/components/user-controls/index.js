import React from 'react'
import './user-controls.less'
import { connect } from "react-redux";
import { Link } from 'react-router-dom'

export class UserControls extends React.Component {

    render = () => 
        { if (localStorage.logged_in === 'true') { 
            return <div>
                <Link className="register-login" to="/profile" title="Личный кабинет">
                    <img src={require('../../images/registration.png')} alt='profile' />
                    Личный кабинет
                </Link>
                <Link className="register-login" to="/logout" title="Выйти">
                    <img src={require('../../images/login.png')} alt='logout' />
                    Выйти
                </Link>
            </div>
        }
        if (localStorage.logged_in !== 'true') { return <div>
            <Link className="register-login" to="/registration" title="Регистрация">
                <img src={require('../../images/registration.png')} alt='registration' />
                Регистрация
            </Link>
            <Link className="register-login" to="/login" title="Войти">
                <img src={require('../../images/login.png')} alt='login' />
                Войти
            </Link>
        </div>

        }
        }
    }

const mapStateToProps = (state) => ({
    user_info: state.user_info,
})

export default connect(mapStateToProps)(UserControls)