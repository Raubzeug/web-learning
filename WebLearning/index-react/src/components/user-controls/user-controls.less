@import "../../styles/library.less";

.register-login {
    .non-decorated-link(@light-black-text);
    text-align: right;
    padding: 5px;

    img {
        max-width: 20px;
    }
}