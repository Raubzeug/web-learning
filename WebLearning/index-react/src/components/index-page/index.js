import React from 'react';
import Company from '../Company'
import Course from '../Course';
import Review from '../Review'
import './index-page.less'
import Pic1 from '../../images/python-1.png'
import Pic2 from '../../images/css-1.png'
import Wolf from '../../images/wolf.png'
import Rabbit from '../../images/rabbit.png'
import Bear from '../../images/bear.png'

const COURSES = [
    {
        name: 'Web разработка на Python',
        lang: 'Python',
        img: Pic1,
        caption: 'Много всего интересного про язык программирования Python',
        url: '/courses/python-1.html'
    },
    {
        name: 'Верстка с помощью CSS',
        lang: 'CSS',
        img: Pic2,
        caption: `Всякая разная информация о верстке с помощью CSS и еще всякая
         разная информация и еще и еще и еще чтобы проверить, как верстается карточка курса.`,
        url: '/courses/css-1.html'
    },
]

const REVIEWS = [
    {text: 'Супер!', author: 'Елена'},
    {text: 'Отлично!', author: 'Елена'},
    {text: 'Отзыв немного подлинее, чтобы проверить форматирование!', author: 'Елена'},
    {text: 'Отзыв немного подлинее, чтобы проверить форматирование!', author: 'Елена'}
]

const PARTNERS = [
    {name: 'АО "Волчий софт"', img: Wolf},
    {name: 'Государственная корпорация "Зайцы-тестировщики"', img: Rabbit},
    {name: 'Студия "Дизайн от медведя Семена"', img: Bear}
]

const IndexContent = () => (
    <section className="content content__two-columns">
        <div className="content-left-wrapper">
            <div className="content__block">
                <div className="content__header">
                    Популярные курсы
                </div>
                <div className="courses">
                    {COURSES.map((element, index) => (
                        <Course key={index} course={element} />
                    )
                    )}
                </div>
            </div>
            <div className="feedback__block">
                <div className="content__header">Отзывы студентов</div>
                {REVIEWS.map((element, index) => (
                    <Review key={index} review={element} />
                )
                )}
            </div>
        </div>
        <aside className="content-right-wrapper">
            <div className="content__header">Предложения работодателей</div>
            {PARTNERS.map((element, index) => (
            <Company key={index} name={element.name} img={element.img} />
            )
            )}
        </aside>

    </section>
)

export default IndexContent