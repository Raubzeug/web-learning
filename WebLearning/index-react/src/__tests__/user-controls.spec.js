import React from 'react'
import { mount } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import toJson from 'enzyme-to-json'
import { UserControls } from '../components/user-controls'


describe('UserControls test', () => {
    it('should render properly NOT logged_in', () => {
        const wrapper = mount(
            <Router>
                <UserControls />
            </Router>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
        wrapper.unmount()
    })
    it('should render properly logged_in', () => {
        localStorage.setItem('logged_in', true)
        const wrapper = mount(
            <Router>
                <UserControls />
            </Router>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
        wrapper.unmount()
    })
})