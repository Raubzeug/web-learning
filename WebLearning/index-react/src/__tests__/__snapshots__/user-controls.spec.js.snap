// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserControls test should render properly NOT logged_in 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UserControls>
      <div>
        <Link
          className="register-login"
          title="Регистрация"
          to="/registration"
        >
          <a
            className="register-login"
            href="/registration"
            onClick={[Function]}
            title="Регистрация"
          >
            <img
              alt="registration"
              src="registration.png"
            />
            Регистрация
          </a>
        </Link>
        <Link
          className="register-login"
          title="Войти"
          to="/login"
        >
          <a
            className="register-login"
            href="/login"
            onClick={[Function]}
            title="Войти"
          >
            <img
              alt="login"
              src="login.png"
            />
            Войти
          </a>
        </Link>
      </div>
    </UserControls>
  </Router>
</BrowserRouter>
`;

exports[`UserControls test should render properly logged_in 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UserControls>
      <div>
        <Link
          className="register-login"
          title="Личный кабинет"
          to="/profile"
        >
          <a
            className="register-login"
            href="/profile"
            onClick={[Function]}
            title="Личный кабинет"
          >
            <img
              alt="profile"
              src="registration.png"
            />
            Личный кабинет
          </a>
        </Link>
        <Link
          className="register-login"
          title="Выйти"
          to="/logout"
        >
          <a
            className="register-login"
            href="/logout"
            onClick={[Function]}
            title="Выйти"
          >
            <img
              alt="logout"
              src="login.png"
            />
            Выйти
          </a>
        </Link>
      </div>
    </UserControls>
  </Router>
</BrowserRouter>
`;
